game/graphics/gfx.h

```diff
@@ -122,61 +106,59 @@ struct GfxGlobalSettings {
 namespace Gfx {
 
 extern GfxGlobalSettings g_global_settings;
-extern GfxSettings g_settings;
-
-struct DebugSettings {
-  bool show_imgui = false;
-  bool ignore_imgui_hide_keybind = false;
-  std::vector<DebugTextFilter> debug_text_filters = {};
-  bool debug_text_check_range = false;
-  float debug_text_max_range = 0;
-
-  void load_settings(const ghc::filesystem::path& filepath);
-  void save_settings();
-};
-
-extern DebugSettings g_debug_settings;
+extern GameSettings::DebugSettings g_debug_settings;
 
 const GfxRendererModule* GetCurrentRenderer();
 
 u32 Init(GameVersion version);
 void Loop(std::function<bool()> f);
 u32 Exit();
 
-Pad::MappingInfo& get_button_mapping();
-
 u32 vsync();
 void register_vsync_callback(std::function<void()> f);
 void clear_vsync_callback();
 u32 sync_path();
 void send_chain(const void* data, u32 offset);
 void texture_upload_now(const u8* tpage, int mode, u32 s7_ptr);
 void texture_relocate(u32 destination, u32 source, u32 format);
-void set_levels(const std::vector<std::string>& levels);
+/// Unused
 void poll_events();
+void set_levels(const std::vector<std::string>& levels);
+
+// InputMonitor usages
+std::optional<std::shared_ptr<PadData>> get_current_frames_pad_data(const int port);
+int update_rumble(const int port, const u8 low_intensity, const u8 high_intensity);
+std::pair<s32, s32> get_mouse_pos();
+int get_controller_count();
+std::string get_controller_name(const int id);
+void set_controller_id_for_port(const int id, const int port);
+void set_keyboard_enabled(const bool enabled);
+void set_mouse_enabled(const bool enabled);
+void ignore_background_controller_events(const bool ignore);
+void set_controller_led(const int port, const u8 red, const u8 green, const u8 blue);
+
+// VideoMonitor usages
 u64 get_window_width();
 u64 get_window_height();
-void set_window_size(u64 w, u64 h);
+void set_fullscreen_display(u64 display_id);
+void set_window_size(u64 width, u64 hheight);
 void get_window_scale(float* x, float* y);
-std::tuple<double, double> get_mouse_pos();
-GfxDisplayMode get_fullscreen();
-int get_screen_vmode_count();
-int get_screen_rate(s64 vmode_idx);
-int get_monitor_count();
-void get_screen_size(s64 vmode_idx, s32* w, s32* h);
+int get_connected_display_count();
+std::string get_connected_display_name(int id);
+int get_active_display_refresh_rate();
+void get_active_display_size(s32* w, s32* h);
+void set_window_resizable(bool resizable);
+WindowDisplayMode get_window_display_mode();
+void set_window_display_mode(WindowDisplayMode mode);
+
+// Global Settings Related
+// TODO - how many of these actually need to be globals and instead should just be virtual functions
+// in the gfx pipeline
```

Comment:
Some have to be globals so they can be called from GOAL.
That said, I think there's still a lot of opportunity to clean this up to be less confusing (there's the globals, the struct of function pointers, and the virtual `Gfx` thing - I think we don't need both)